datafusion/core/src/sql/parser.rs

```diff
@@ -277,6 +308,12 @@ impl<'a> DFParser<'a> {
 
         let has_header = self.parse_csv_has_header();
 
+        let has_partition = self.parse_has_partition();
+        let mut table_partition_cols: Vec<String> = vec![];
+        if has_partition {
+            table_partition_cols = self.parse_partitions()?;
+        }
```

Comment:
```suggestion
        let table_partition_cols = if self.parse_has_partition() {
          self.parse_partitions()?;
        } else {
          vec![]
        };
```

If you wanted to avoid a `mut`

Reply:
thanks, that's nice